router/api.go

```diff
@@ -204,6 +209,107 @@ func (api *API) DeleteRoute(ctx context.Context, w http.ResponseWriter, req *htt
 	w.WriteHeader(200)
 }
 
+func (api *API) CreateCert(ctx context.Context, w http.ResponseWriter, req *http.Request) {
+	log, _ := ctxhelper.LoggerFromContext(ctx)
+
+	var cert *router.Certificate
+	if err := json.NewDecoder(req.Body).Decode(&cert); err != nil {
+		log.Error(err.Error())
+		httphelper.Error(w, err)
+		return
+	}
+
+	l := api.router.HTTP.(*HTTPListener)
+	err := l.AddCert(cert)
+	if err != nil {
+		rjson, jerr := json.Marshal(&cert)
+		if jerr != nil {
+			log.Error(jerr.Error())
+			httphelper.Error(w, jerr)
+			return
+		}
+		jsonError := httphelper.JSONError{Detail: rjson}
```

Comment:
Is there a reason to provide the request JSON in the detail? Do we do it anywhere else? It may not be accurate because a round-trip through the decoder/encoder is not guaranteed to provide the exact same bytes.